### PR TITLE
Use self.fig.tight_layout to avoid cut-off labels

### PR DESCRIPTION
**Pull request by rebeccabilbro**
This PR addresses #739, adding `self.fig.tight_layout` to the `finalize` method of several visualizers to ensure that the labels are not cut off when plotted.

I have made the following changes:

1. Added `self.fig.tight_layout()` to `ConfusionMatrix` and `PosTagVisualizer`
2. Updated `plt.tight_layout` to `self.fig.tight_layout` in `ClassificationReport`, `JointPlot`, and `FeatureImportances`, leveraging the `fig` attribute introduced to our API in #936. 
3. Updated the baseline images for one confusion matrix test and two postag tests 

### Sample Code and Plot

The `ConfusionMatrix` plot in our docs now looks like this:
Screen Shot 2020-04-08 at 5 15 51 PM

And the `PosTagVisualizer` looks like:
Screen Shot 2020-04-08 at 5 15 16 PM

### CHECKLIST

- [x] _Is the commit message formatted correctly?_
- [ ] _Have you noted the new functionality/bugfix in the release notes of the next release?_
- [x] _Included a sample plot to visually illustrate your changes?_
- [x] _Do all of your functions and methods have docstrings?_
- [x] _Have you added/updated unit tests where appropriate?_
- [x] _Have you updated the baseline images if necessary?_
- [x] _Have you run the unit tests using `pytest`?_
- [x] _Is your code style correct (are you using PEP8, pyflakes)?_
- [ ] _Have you documented your new feature/functionality in the docs?_
- [x] _Have you built the docs using `make html`?_

### CODE REVIEW

File: docs/api/classifier/confusion_matrix.rst [hunk 1]
```diff
@@ -68,8 +68,9 @@ Class names can be added to a ``ConfusionMatrix`` plot using the ``label_encoder
     :alt: ConfusionMatrix plot with class names
 
     from sklearn.datasets import load_iris
-    from sklearn.model_selection import train_test_split as tts
     from sklearn.linear_model import LogisticRegression
+    from sklearn.model_selection import train_test_split as tts
```
Comment (bbengfort) [inline docs/api/classifier/confusion_matrix.rst:72]: :-)

File: tests/test_text/test_postag.py [hunk 2]
```diff
@@ -229,7 +230,8 @@ def test_frequency_mode(self):
         # Assert that ticks are set properly
         assert ticks_ax == sorted_tags
 
-        self.assert_images_similar(ax=ax, tol=0.5)
+        # Fails on Miniconda/Appveyor with images not close (RMS 5.302)
+        self.assert_images_similar(ax=ax, tol=5.5)
```
Comment (bbengfort) [inline tests/test_text/test_postag.py:234]: I understand that updating the tolerance globally was required for both Travis and AppVeyor because Travis tests miniconda; however if you'd like to keep the more specific python-only test, we could use the is_windows_or_conda functionality that we built in for this reason:

```python
from tests.base import IS_WINDOWS_OR_CONDA

tol = 5.5 if IS_WINDOWS_OR_CONDA else 0.25
```

But it's good to note that that the tight layout has variable behavior!

File: yellowbrick/classifier/classification_report.py [hunk 3]
```diff
@@ -283,7 +283,7 @@ def finalize(self, **kwargs):
         self.ax.set_xticklabels(self._displayed_scores, rotation=45)
         self.ax.set_yticklabels(self.classes_)
 
-        plt.tight_layout()  # TODO: Could use self.fig now
+        self.fig.tight_layout()
```
Comment (bbengfort) [inline yellowbrick/classifier/classification_report.py:286]: Our first use of `self.fig` in earnest! w00t!

File: tests/test_text/test_postag.py (hunk 2 above)
Comment (rebeccabilbro) [reply tests/test_text/test_postag.py]: Ah, thank you for reminding me about this utility; I've updated the tests accordingly.
